libnavigation-ui/src/test/java/com/mapbox/navigation/ui/route/MapRouteProgressChangeListenerTest.java

```diff
@@ -1,57 +0,0 @@
-package com.mapbox.navigation.ui.route;
-
-import com.mapbox.api.directions.v5.models.DirectionsRoute;
-import com.mapbox.navigation.base.trip.model.RouteProgress;
-
-import org.junit.Test;
-
-import edu.emory.mathcs.backport.java.util.Collections;
-
-import static org.mockito.ArgumentMatchers.eq;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.verifyZeroInteractions;
-import static org.mockito.Mockito.when;
-
-public class MapRouteProgressChangeListenerTest {
```

Comment:
these tests didn't include a positive case. there's nothing testing when a route is actually changed.

so this pull request 
1. migrates this test to kotlin
2. to mockk
3. adds the null test case that is crashing
4. adds a positive test case